### DISCUSSION TITLE
Welcome to wgpu4k Discussions!

### DISCUSSION THREAD

**Discussion by ygdrasil-io** [wgpu4k/wgpu4k#31] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by hamoid** [comment]
Hi :-) I'm an artist creating software based audiovisual works. I use and contribute to [OPENRNDR](https://github.com/openrndr/), a multiplatform Kotlin-based creative coding framework. It uses LWJGL on the desktop, so running programs on Android is not currently possible. It does run on a web browser, but not yet compute shaders. Those are two features that would be nice to have in the future, and I wonder if wgpu might help. I'm here because I'm curious about anything related to graphics and Kotlin :)

**Reply by ygdrasil-io** [reply]
@hamoid Welcome :)

This sample shows how to use compute shaders:
https://github.com/wgpu4k/wgpu4k/blob/main/examples/common/src/commonMain/kotlin/io.ygdrasil.wgpu.examples/scenes/graphics.techniques/Particles.kt

This is not yet working on Firefox and Desktop (which is using the same backend as Firefox), but on Chrome, everything is alright. Firefox plans to enable WebGPU by default at the end of the year, so we can expect a lot of bug fixes hopefully.

On Android, this is currently more in the thinking stage rather than work in progress. I guess I will first make it work on iOS, but that's on my to-do list. If you want to work even a little on the subject, there is some Rust code to write to create the JNI layer and then use it on Kotlin Android. I found this project that could be a good start: https://github.com/ruffle-rs/ruffle-android

**Reply by hamoid** [reply]
Thank you :-) I just saw you released https://github.com/wgpu4k/hello-cube Exciting!

ps. Rust I haven't tried yet. Maybe some day :-)
